transaction.go

```diff
@@ -96,6 +98,153 @@ func (tx *Transaction[T]) GetSignedTransactionBodyBytes(transactionIndex int) []
 	return tx.signedTransactions._Get(transactionIndex).(*services.SignedTransaction).GetBodyBytes()
 }
 
+// Helper function to cast any type of transaction to the base transaction
+func castFromAnyToBaseTransaction(tx any) (*Transaction[TransactionInterface], error) { // nolint
```

Comment:
If all transaction type cases call `castFromConcreteToBaseTransaction(t)` why do we have this switch case?